src/main/java/bio/terra/janitor/service/cleanup/flight/GoogleDatasetCleanupStep.java

```diff
@@ -0,0 +1,42 @@
+package bio.terra.janitor.service.cleanup.flight;
+
+import bio.terra.cloudres.common.ClientConfig;
+import bio.terra.cloudres.google.bigquery.BigQueryCow;
+import bio.terra.generated.model.CloudResourceUid;
+import bio.terra.janitor.db.JanitorDao;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryException;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.DatasetId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Step to cleanup Google BigQuery Dataset resource. */
+public class GoogleDatasetCleanupStep extends ResourceCleanupStep {
+  private final Logger logger = LoggerFactory.getLogger(GoogleDatasetCleanupStep.class);
+  private final BigQueryCow bigQueryCow;
+
+  public GoogleDatasetCleanupStep(ClientConfig clientConfig, JanitorDao janitorDao) {
+    super(clientConfig, janitorDao);
+    this.bigQueryCow = new BigQueryCow(clientConfig, BigQueryOptions.getDefaultInstance());
+  }
+
+  @Override
+  protected StepResult cleanUp(CloudResourceUid resourceUid) {
+    DatasetId datasetId =
+        DatasetId.of(
+            resourceUid.getGoogleBigQueryDatasetUid().getProjectId(),
+            resourceUid.getGoogleBigQueryDatasetUid().getDatasetId());
+    try {
+      // Delete dataset with content inside.
```

Comment:
nit: Comment feels unnecessary since the next line of code is very self-descriptive